Library/Formula/mlkit.rb

```diff
@@ -0,0 +1,43 @@
+class Mlkit < Formula
+  desc "Compiler for the Standard ML programming language"
+  homepage "https://melsman.github.io/mlkit"
+  url "https://github.com/melsman/mlkit/archive/mlkit-4.3.9.tar.gz"
+  sha256 "3c6adbeb9a85f7b3586d0961fd3b170ff31e09fa0ff12889b76b9ceb459059c4"
+
+  depends_on "autoconf" => :build
+  depends_on "mlton" => :build
+  depends_on "gmp"
+
+  def install
+    system "./autobuild"
+    system "./configure", "--prefix=#{prefix}"
+
+    # The ENV.permit_arch_flags specification is needed on 64-bit
+    # machines because the mlkit compiler generates 32-bit machine
+    # code whereas the mlton compiler generates 64-bit machine
+    # code. Because of this difference, the ENV.m64 and ENV.m32 flags
+    # are not sufficient for the formula as clang is used by both
+    # tools in a single makefile target. For the mlton-compilation of
+    # sml-code, no arch tags are used for the clang assembler
+    # invocation. Thus, on a 32-bit machine, both the mlton-compiled
+    # binary (the mlkit compiler) and the 32-bit native code generated
+    # by the mlkit compiler will be running 32-bit code.
+
+    ENV.permit_arch_flags if MacOS.prefer_64_bit?
+    system "make", "mlkit"
+    system "make", "mlkit_libs"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.sml").write <<-EOS.undent
+      fun f(x) = x + 2
+      val a = [1,2,3,10]
+      val b = List.foldl (op +) 0 (List.map f a)
+      val res = if b = 24 then "OK" else "ERR"
+      val () = print ("Result: " ^ res ^ "\\n")
+    EOS
+    system "#{bin}/mlkit", "-o", "test", "test.sml"
+    system "./test"
```

Comment:
I wrote the test according to the guide - it is simple and is similar to
the tests found in other packages. It writes "Result: OK" on stdout if it
succeeds.

Den fredag den 1. april 2016 skrev ilovezfs <notifications@github.com>:

> In Library/Formula/mlkit.rb
> <https://github.com/Homebrew/homebrew/pull/49933#discussion_r58202510>:
>
> > +    ENV.permit_arch_flags if MacOS.prefer_64_bit?
> > +    system "make", "mlkit"
> > +    system "make", "mlkit_libs"
> > +    system "make", "install"
> > +  end
> > +
> > +  test do
> > +    (testpath/"test.sml").write <<-EOS.undent
> > +      fun f(x) = x + 2
> > +      val a = [1,2,3,10]
> > +      val b = List.foldl (op +) 0 (List.map f a)
> > +      val res = if b = 24 then "OK" else "ERR"
> > +      val () = print ("Result: " ^ res ^ "\\n")
> > +    EOS
> > +    system "#{bin}/mlkit", "-o", "test", "test.sml"
> > +    system "./test"
>
> Anything we're expecting this test to do other than execute without
> blowing up?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/49933/files/b3e5bbd768c42d8b576ce685f5a8926f469733d8#r58202510>
>